### PR TITLE
Jeffrey>Tzu Ning correct fix letter code dropdown 

### PR DESCRIPTION
**Pull request by JeffLi117**
# Description
image

## Related PRS (if any):
none
…

## Main changes explained:
- Add a handler function that fires when the new team code has been accepted and sent to database
- Said handler function updates state (teamCodes, selectedCodes, and summaries) within WeeklySummariesReport.jsx to accurately reflect the changes within that would occur if updated data were pulled from the database, but without refreshing

## How to test:
1. check into current branch
2. do `npm install` and `...` to run this PR locally
3. log in as Admin/Owner user
4. go to Other Links → Reports → Weekly Summaries Reports
5. select a team code from the dropdown that has a label of (1)
6. change the team's code (input element next to Media URL) to a code that does not exist yet
7. ensure that the initial team code from step 5 is not present within the dropdown menu
8. click on the dropdown again and see if that new team code is reflected in the dropdown menu
9. select that team code to ensure the correct number of members are populated on the screen
10. select another team code
11. change the team's code to a code that DOES exist already
12. click on the dropdown and check that the team code from step 11 is updated correctly
13. select that team code to ensure the correct number of members are populated on the screen
14. refresh page to confirm that the changes you made previously (and saw accurately reflected on the front-end before the refresh) are the same after you refresh

**Note** - this is where you can find the team code (example below is C-SOC):
image

## Screenshots or videos of changes:

Before change: 
https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/112568110/5617d750-d275-40d3-8a95-f787b5a30c61

After change:
https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/112568110/3b30d14c-2207-455a-819d-2fc9be638131

### CODE REVIEW

**Comment by olenadanykh** [comment]
Hi @JeffLi117, I've tested your PR, and I didn't experience any errors. Everything works as expected on my end. The total number of team members updates smoothly without needing a manual page refresh, proving effective for both existing and newly created teams. Great job!

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/49422649/a288ee39-c2ad-47d2-8bcb-7fb86c82f0ed

**Review by olenadanykh — DISMISSED** [review]

**Review by sxiong5 — COMMENTED** [review]
Hey @JeffLi117, I have tested the PR and everything works very great!! Nice job!

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/62367016/39f340e0-cb84-4c30-a44b-0120feef22f1

**Review by pika-chu11 — DISMISSED** [review]
Hi @JeffLi117 , I followed your instruction of how to test and all things works as expected! Good Job!

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/67128170/5841581b-eee2-4772-9297-e18cb2ad4109

**Comment by beblicarl** [comment]
I am not sure what I am doing wrongly but I keep getting this error when I change it manually

image

**Review by KurtisIvey — DISMISSED** [review]
Great job. This was the original implementation of effect I had in mind when you brought this up before.

You may want to clean up the console.logs you used while developing and some of the old code you've commented out, but it's all functioning perfectly. 

Changed team code on owner acct
Screenshot 2023-11-17 at 10 58 47 AM


Data persists on admin acct
Screenshot 2023-11-17 at 10 57 32 AM

**Review by ptrpengdev — CHANGES_REQUESTED** [review]
Hi Jeffery, I have tested your code locally, and most functionalities work as expected. However, I did find one issue with the Select Team Code dropdown bar, which displayed incorrect data after modifying the team code.

**Issue:**
- Search Bar "Select Team Code" Update incorrectly

**Env Setup:**
- Frontend: [Jeffrey_newfix_letter_code_dropdown](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/pull/1545)
- Backend: development (latest)

**Step To Reproduce:**
- Log in with Admin privileges and navigate to the Weekly Summaries Report.
- Select multiple team codes, ensuring to include at least one team code with only one team member.
- Modify the team code for the member who is in a one-person team and update it.
- Check the "Select Team Code" dropdown bar; incorrect data is displayed.

**Video:**
In the following video, three team codes are selected: A-AAA (3), A-BBA (1), A-BBB (1). The team code A-BBB has only one person. After updating the team code for the sole member in A-BBB, the dropdown bar should display A-AAA (4), A-BBA (1). However, it currently displays A-BBA (1), A-BBB (1).

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/69882989/4b5a0a5a-fdfb-4cfd-99ec-0c69a88111f4

**Review by KavyaAlla — DISMISSED** [review]
Hi @JeffLi117 I have tested this PR locally and everything is working as expected. The new team codes created or edited to existing codes are being reflected with out refreshing. Also the changes made are visible after refreshing. Good job!!
Details are in below Video.

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/61841929/d4a6babf-4728-400a-b2d5-3d99b8f995b8

File: src/components/WeeklySummariesReport/FormattedReport.jsx [hunk 1]
```diff
@@ -277,15 +281,17 @@ function WeeklySummaryMessage({ summary, weekIndex }) {
   );
 }
 
-function TeamCodeRow({ canEditTeamCode, summary }) {
+function TeamCodeRow({ canEditTeamCode, handleTeamCodeChange, summary }) {
   const [teamCode, setTeamCode] = useState(summary.teamCode);
+  const [_teamCode, set_TeamCode] = useState(teamCode);
```
Comment (shalabhtayal) [inline src/components/WeeklySummariesReport/FormattedReport.jsx:286]: the state name "_teamCode" runs parallel to original "teamCode" and could be very confusing

Comment (shalabhtayal) [inline src/components/WeeklySummariesReport/FormattedReport.jsx:286]: the state name "_teamCode" runs parallel to original "teamCode" and could be very confusing

**Review by shalabhtayal — APPROVED** [review]
The code does work, clear variable names and some more comments would be helpful.

Also, how the data consistency is maintained with concurrent users since these updates happen on the client side ? Would database update in the backend (by another user) update the list in real time for all live clients (similar to meteor) ?

**Review by Alforoan — CHANGES_REQUESTED** [review]
I am getting a 403 error.

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/118868270/6bc69a90-c6cb-4273-aedf-fecceb8d2ecd

**Comment by JeffLi117** [comment]
> I am getting a 403 error.
> 
>  2023-11-29.21-22-18.mp4

Please ensure your HGNRest development branch is up to date. I was receiving a 500 error before I updated my back-end (see below). 
Please let me know if your issue still persists after updating!

image

**Comment by JeffLi117** [comment]
> I am not sure what I am doing wrongly but I keep getting this error when I change it manually
> 
> image

Please ensure your HGNRest development branch is up to date. I was receiving a 500 error before I updated my back-end (see below). 
Please let me know if your issue still persists after updating!

image

**Comment by JeffLi117** [comment]
> Hi Jeffery, I have tested your code locally, and most functionalities work as expected. However, I did find one issue with the Select Team Code dropdown bar, which displayed incorrect data after modifying the team code.
> 
> **Issue:**
> 
> * Search Bar "Select Team Code" Update incorrectly
> 
> **Env Setup:**
> 
> * Frontend: [Jeffrey_newfix_letter_code_dropdown](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/pull/1545)
> * Backend: development (latest)
> 
> **Step To Reproduce:**
> 
> * Log in with Admin privileges and navigate to the Weekly Summaries Report.
> * Select multiple team codes, ensuring to include at least one team code with only one team member.
> * Modify the team code for the member who is in a one-person team and update it.
> * Check the "Select Team Code" dropdown bar; incorrect data is displayed.
> 
> **Video:** In the following video, three team codes are selected: A-AAA (3), A-BBA (1), A-BBB (1). The team code A-BBB has only one person. After updating the team code for the sole member in A-BBB, the dropdown bar should display A-AAA (4), A-BBA (1). However, it currently displays A-BBA (1), A-BBB (1).
> 
>  BUG.mp4

Hi @ptrpengdev, please re-test my PR with the latest changes. I believe I had an error in which I was using .find instead of .findIndex for the selectedCodes state. It should now work properly. Thank you for pointing this issue out!

**Review by keyunhuangg — APPROVED** [review]
1) Checkout of this brand and logged in as owner account
2) Tested changing one team member's team code to a team code that contains more than 1
3) Tested changing one team member's team code to a team code that does not exist

Screenshot 2023-11-30 at 4 35 14 PM
Screenshot 2023-11-30 at 4 35 28 PM

**Review by sucheta90 — COMMENTED** [review]
@JeffLi117 I ran the local branch, followed the steps to test and logged in as Admin, however I couldn't edit the Team Code, it was not an Input field as it should be (as per the description). Please refer to the video.

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/101291283/e05bba41-7449-4773-9cbd-a93e82a3db82

**Comment by DragonSlayer2401** [comment]
I tested your PR with an Admin and Owner account. The admin account was unable to edit the team codes. WIth the owner account I was able to edit the team codes and everything worked as intended. Please watch the below video for more details:


https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/102876103/3b803c1d-17b8-4053-975e-f4c175cbf789

**Review by Alforoan — COMMENTED** [review]
Everything works as intended. Nice!

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/118868270/c0d875e1-31be-455a-8987-af2db8e70f0b

**Comment by JeffLi117** [comment]
@DragonSlayer2401  @sucheta90 
I spoke with Jae - Admin accounts should now have the same ability to test this PR. Please re-test and let me know if the changes work as I intended.
I'll be sure to update this in the PR description!

**Comment by Haoxiang310** [comment]
I've tested your PR with owner and admin role. 
For the owner role, everything works as expected:

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/135764868/44514ae4-d2d4-4586-9ed7-1b0e5c381cf1

But for the admin role, when I tried to save the new team code change, it had an error with 403:

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/135764868/5ad84e70-b16a-40b8-808b-de7fbdf59b6b

**Comment by yuhuimin1996** [comment]
Hi @JeffLi117 ,
I've tested this PR using both admin user and owner user. Both of them can work well as you described, good job! 
The owner user test video:

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/108281570/6465791d-5fa3-4e3d-a2bb-e97efa4c21b9
The admin user test video:

Uploading Screen Recording 2024-01-11 at 18.48.14.mov…

**Review by Renanluizssx — COMMENTED** [review]
Hi @JeffLi117, i don't what i' am doing wrong, but when i put the team code, it appears the total numbers correctly, however when i put the team code manually, nothing happens, see my review below:

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/91035340/73bea5fe-d5c3-48e3-b5d2-f73772375f72

**Comment by one-community** [comment]
Thank you all, merging!

**Pull request merged**